lib/puppet/provider/ios_radius_global/cisco_ios.rb

```diff
@@ -66,7 +66,12 @@ def update(context, _name, should)
     end
   end
 
-  def canonicalize(_context, resources)
-    resources
+  def canonicalize(context, resources)
+    resources.each do |resource|
+      current_instance = (get(context) || []).find { |key| key[:name] == resource[:name] }
+      if resource[:attributes] == []
+        resource.delete(:attributes) unless current_instance && current_instance[:attributes]
```

Comment:
What's the deal here? There is a significant difference in `nil` and `[]` when coming from the user: the `nil` means "don't manage", the empty array means "delete everything". This change doesn't seem to respect that - unless I've missed something else in the code.

Reply:
So `attributes => []` seems to be the handling to `unset` the `attributes` from the device, so in a manifest if there is:

```
ios_radius_global { 'default':
  attributes => []
}
```

And there are attributes present on the device then it will `unset` them, but on the next run with the above manifest then it will think it needs to change attributes - so this check is to remove `attributes` if it is `[]` and only if there is currently attributes on the current state of the device. So on first run with `[]` then it will remove them as expected, and on second run it won't have to do anything as there already is no attributes on the device.

Reply:
If there are no attributes then yes it will be `nil`, having `resource[:attributes] ||= is[:attributes]` I think would cause more issues as whatever the manifest is looking to manage will be overwritten by the current state so no changes will ever occur and the current state will always be kept even if the user wants to perform a change?

Reply:
Or do you mean just return the `[]` in the `get` and ignore the `canonicalize`?